src/Interpreters/IcebergMetadataLog.cpp

```diff
@@ -106,4 +107,37 @@ void insertRowToLogTable(
             .metadata_content = row,
             .row_in_file = row_in_file});
 }
+
+String dumpMetadataObjectToString(const Poco::JSON::Object::Ptr & metadata_object)
+{
+    std::ostringstream oss; // STYLE_CHECK_ALLOW_STD_STRING_STREAM
+    Poco::JSON::Stringifier::stringify(metadata_object, oss);
+    return removeEscapedSlashes(oss.str());
+}
+
+void insertRowToLogTable(
```

Comment:
Do you still need the "old" overload? If so, add a comment explaining the difference between both overloads..

And btw, is `dumpMetadataObjectToString` that expensive that you need to optimize it in a few code paths?

Reply:
In other places `insertRowToLogTable` called with strings from other sources like https://github.com/Altinity/ClickHouse/blob/antalya-25.8/src/Storages/ObjectStorage/DataLakes/Iceberg/AvroForIcebergDeserializer.cpp#L119.
It is also JSON string, but with some complex procedure to get.

In my test on my desktop with ~7k parquet files `metadata.json` has size near 5Mb.
Query ```select count() from iceberg.table``` does not read data, gets size from metadata only.
Speed changed from 3.5 seconds to 0.3 seconds.
JSON parsing is fast, but serialization back to string isn't.

Reply:
But getting json after condition check make sense in all case. Changed for all.